autogpt_platform/backend/backend/data/credit.py

```diff
@@ -328,7 +328,7 @@ async def spend_credits(
 
         # Auto top-up if balance just went below threshold due to this transaction.
         auto_top_up = await get_auto_top_up(user_id)
-        if balance < auto_top_up.threshold <= balance - cost:
+        if balance < auto_top_up.threshold <= balance + cost:
```

Comment:
I've to admit this is unrelated, I encountered this bug when testing the USD number change.